sway/commands.c

```diff
@@ -1003,145 +1010,226 @@ static struct cmd_results *cmd_mouse_warping(int argc, char **argv) {
-		// move scratchpad ...
-		if (view->type != C_CONTAINER && view->type != C_VIEW) {
-			return cmd_results_new(CMD_FAILURE, "move scratchpad", "Can only move containers and views.");
-		}
-		swayc_t *view = get_focused_container(&root_container);
-		int i;
-		for (i = 0; i < scratchpad->length; i++) {
-			if (scratchpad->items[i] == view) {
-				hide_view_in_scratchpad(view);
-				sp_view = NULL;
-				return cmd_results_new(CMD_SUCCESS, NULL, NULL);
-			}
-		}
-		list_add(scratchpad, view);
-		if (!view->is_floating) {
-			destroy_container(remove_child(view));
+        return cmd_results_new(CMD_SUCCESS, NULL, NULL);
+}
+
+static struct cmd_results *cmd_move_container_to_output(int argc, char **argv) {
+	struct cmd_results *error = NULL;
+	if ((error = checkarg(argc, "move container/window", EXPECTED_AT_LEAST, 1))) {
+		return error;
+	}
+
+	swayc_t *view = get_focused_container(&root_container);
+	if (view->type != C_CONTAINER && view->type != C_VIEW) {
+		return cmd_results_new(CMD_FAILURE, "move", "Can only move containers and views.");
+	}
+
+	swayc_t *output = NULL;
+	struct wlc_point abs_pos;
+	get_absolute_center_position(view, &abs_pos);
+	if (!(output = output_by_name(argv[0], &abs_pos))) {
+		return cmd_results_new(CMD_FAILURE, "move",
+				       "Can't find output with name/direction '%s' @ (%i,%i)", argv[0], abs_pos.x, abs_pos.y);
+	} else {
+		swayc_t *container = get_focused_container(output);
+		if (container->is_floating) {
+			move_container_to(view, container->parent);
 		} else {
-			remove_child(view);
-		}
-		wlc_view_set_mask(view->handle, 0);
-		arrange_windows(swayc_active_workspace(), -1, -1);
-		swayc_t *focused = container_under_pointer();
-		if (focused == NULL) {
-			focused = swayc_active_workspace();
+			move_container_to(view, container);
 		}
-		set_focused_container(focused);
-	} else if (strcasecmp(argv[0], "position") == 0 && strcasecmp(argv[1], "mouse") == 0) {
-		if (view->is_floating) {
-			swayc_t *output = swayc_parent_by_type(view, C_OUTPUT);
-			struct wlc_geometry g;
-			wlc_view_get_visible_geometry(view->handle, &g);
-			const struct wlc_size *size = wlc_output_get_resolution(output->handle);
+	}
+
+        return cmd_results_new(CMD_SUCCESS, NULL, NULL);
+}
 
-			struct wlc_point origin;
-			wlc_pointer_get_position(&origin);
+static struct cmd_results *cmd_move_container(int argc, char **argv) {
+	struct cmd_results *error = NULL;
+	if ((error = checkarg(argc, "move container/window", EXPECTED_AT_LEAST, 2))) {
+		return error;
+	}
 
-			int32_t x = origin.x - g.size.w / 2;
-			int32_t y = origin.y - g.size.h / 2;
+	if (strcasecmp(argv[0], "to")) {
+		return cmd_results_new(CMD_INVALID, "move", cmd_move_expected_syntax);
+	}
+        /* Skip the 'to' part of the command */
+        argc--;
+        argv++;
+
+	static const struct cmd_handler handlers[] = {
+            {"output", cmd_move_container_to_output},
+            {"workspace", cmd_move_container_to_workspace},
+        };
```

Comment:
Apparently clang-format is not enough. Guess I'll create a local vimrc which I probably should have done long time age.